crates/uv/src/cli.rs

```diff
@@ -834,6 +834,27 @@ pub(crate) struct PipSyncArgs {
     #[arg(long, short = 'C', alias = "config-settings")]
     pub(crate) config_setting: Option<Vec<ConfigSettingEntry>>,
 
+    /// The minimum Python version that should be supported by the requirements (e.g.,
+    /// `3.7` or `3.7.9`).
+    ///
+    /// If a patch version is omitted, the most recent known patch version for that minor version
+    /// is assumed. For example, `3.7` is mapped to `3.7.17`.
+    #[arg(long)]
+    pub(crate) python_version: Option<PythonVersion>,
+
+    /// The platform for which requirements should be installed.
+    ///
+    /// Represented as a "target triple", a string that describes the target platform in terms of
+    /// its CPU, vendor, and operating system name, like `x86_64-unknown-linux-gnu` or
+    /// `aaarch64-apple-darwin`.
+    ///
+    /// WARNING: When specified, uv will select wheels that are compatible with the target platform.
+    /// The resulting environment may not be fully compatible with the current platform. Further,
+    /// distributions that are built from source may ultimately be incompatible with the target
+    /// platform. This option is intended for cross-compilation and other advanced use cases.
```

Comment:
I think this was mentioned in #3111, but how can we provide end users with a list of valid triples they can use? I don't think we should just link to the Rust docs unless we specifically want to guarantee a correspondence there. Basically, my concern here is that users won't know what they're allowed to put here.

Honestly, I have a similar concern with pip's `--platform` flag. Its docs are:

```
  --platform <platform>       Only use wheels compatible with <platform>.
                              Defaults to the platform of the running system.
                              Use this option multiple times to specify
                              multiple platforms supported by the target
                              interpreter.
```

After reading that, I have no clue whatsoever what I'm allowed or even supposed to use as a value for this flag.

(And now I'm wondering what the difference is between pip's `--platform` flag and our `--python-platform` flag...)

Reply:
We already do! Clap shows all the variants automatically along with their Rustdoc.

Reply:
Oh sweet. Perfect.